### PR TITLE
Polish ServletWebRequest and DefaultServerWebExchange

### PR DESCRIPTION
**Pull request by divcon**
1.  Remove unused return value
2. Add Nullable annotaions
3. Simplify if statement

### CODE REVIEW

**Review by sbrannen — CHANGES_REQUESTED** [review]

File: spring-web/src/main/java/org/springframework/web/context/request/ServletWebRequest.java [hunk 1]
```diff
@@ -319,31 +316,22 @@ private void updateResponseStateChanging(String eTag, long lastModifiedTimestamp
 	}
 
 	private boolean validateIfUnmodifiedSince(long lastModifiedTimestamp) {
-		if (lastModifiedTimestamp < 0) {
-			return false;
-		}
 		long ifUnmodifiedSince = parseDateHeader(HttpHeaders.IF_UNMODIFIED_SINCE);
-		if (ifUnmodifiedSince == -1) {
+		if (lastModifiedTimestamp < 0 || ifUnmodifiedSince == -1) {
```
[the remark this answers is not in the document]
Comment (divcon) [reply spring-web/src/main/java/org/springframework/web/context/request/ServletWebRequest.java]: I reverted it.

File: spring-web/src/main/java/org/springframework/web/context/request/ServletWebRequest.java [hunk 2]
```diff
@@ -319,31 +316,22 @@ private void updateResponseStateChanging(String eTag, long lastModifiedTimestamp
...
 		}
 		this.notModified = (ifUnmodifiedSince < (lastModifiedTimestamp / 1000 * 1000));
 		return true;
 	}
 
-	private boolean validateIfModifiedSince(long lastModifiedTimestamp) {
-		if (lastModifiedTimestamp < 0) {
-			return false;
-		}
+	private void validateIfModifiedSince(long lastModifiedTimestamp) {
```
[the remark this answers is not in the document]
Comment (divcon) [reply spring-web/src/main/java/org/springframework/web/context/request/ServletWebRequest.java:330]: I reworked it.

File: spring-web/src/main/java/org/springframework/web/server/adapter/DefaultServerWebExchange.java [hunk 3]
```diff
@@ -278,10 +278,8 @@ else if (validateIfUnmodifiedSince(lastModified)) {
 
 	private boolean validateIfMatch(@Nullable String eTag) {
 		try {
-			if (SAFE_METHODS.contains(getRequest().getMethod())) {
-				return false;
-			}
-			if (CollectionUtils.isEmpty(getRequest().getHeaders().get(HttpHeaders.IF_MATCH))) {
+			if (SAFE_METHODS.contains(getRequest().getMethod())
+					|| CollectionUtils.isEmpty(getRequest().getHeaders().get(HttpHeaders.IF_MATCH))) {
```
[the remark this answers is not in the document]
Comment (divcon) [reply spring-web/src/main/java/org/springframework/web/server/adapter/DefaultServerWebExchange.java]: I reverted it.

**Comment by divcon** [comment]
@sbrannen 
Thank you for your kind review.
I reflected your review.

**Review by sbrannen — APPROVED** [review]

**Comment by sbrannen** [comment]
This has been merged into `main` in c12d93c5d1696fc030b7b2fc0fb999b99795c59c and revised in a832c98cedc56690ad606b53f497814370792d12.

Thanks

**Pull request closed without merging**